jgrapht-core/src/main/java/org/jgrapht/generate/PlantedPartitionGraphGenerator.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * (C) Copyright 2018-2018, by Emilio Cruciani and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+
+package org.jgrapht.generate;
+
+import org.jgrapht.Graph;
+import org.jgrapht.VertexFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * Create a random $l$-planted partition graph.
```

Comment:
Do you by any chance have an external reference where an l-planted partition graph is defined? This would be useful to add. It provides a little more context. Until this PR, I had never heard of a planted partition graph.

Reply:
It is a special case of the Stochastic Block Model (https://en.wikipedia.org/wiki/Stochastic_block_model#Special_cases).

Reply:
Could you add the following in your class description:

```
For more information on planted graphs, refer to:
<ol>
  <li>Condon, A. Karp, R.M. Algorithms for graph partitioning on the planted partition model, Random Structures and Algorithms, Volume 18, Issue 2, p.116-140, 2001</li>
  <li>Fortunato, S. Community Detection in Graphs, Physical Reports Volume 486, Issue 3-5 p. 75-174, 2010</li>
</ol>
```
--

Reply:
I think the class is fine as it is; I would not generalize it further.